### ISSUE TITLE
Compilation of master fails with Qt 5.9 branch

### ISSUE THREAD

**Issue by jgueytat** [KDAB/GammaRay#316]
**Compilation output:**

```
[ 30%] Linking CXX executable ../bin/fontdatabasemodeltest
cd /src/KDAB/GammaRay/build/tests && /usr/bin/cmake -E cmake_link_script CMakeFiles/fontdatabasemodeltest.dir/link.txt --verbose=1
/usr/bin/c++    -fvisibility=hidden -fvisibility-inlines-hidden -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wformat-security -Wno-deprecated -Wextra -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wno-div-by-zero -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fno-operator-names -Wzero-as-null-pointer-constant -O3 -DNDEBUG   CMakeFiles/fontdatabasemodeltest.dir/fontdatabasemodeltest.cpp.o CMakeFiles/fontdatabasemodeltest.dir/__/plugins/fontbrowser/fontdatabasemodel.cpp.o CMakeFiles/fontdatabasemodeltest.dir/__/3rdparty/qt/modeltest.cpp.o CMakeFiles/fontdatabasemodeltest.dir/fontdatabasemodeltest_automoc.cpp.o  -o ../bin/fontdatabasemodeltest -rdynamic /usr/local/Qt-5.9.0/lib/libQt5Widgets.so.5.9.0 /usr/local/Qt-5.9.0/lib/libQt5Test.so.5.9.0 /usr/local/Qt-5.9.0/lib/libQt5Gui.so.5.9.0 /usr/local/Qt-5.9.0/lib/libQt5Core.so.5.9.0 -Wl,-rpath,/usr/local/Qt-5.9.0/lib 
CMakeFiles/fontdatabasemodeltest.dir/__/3rdparty/qt/modeltest.cpp.o : Dans la fonction « ModelTest::data() » :
modeltest.cpp:(.text+0x1fc7) : référence indéfinie vers « bool QTest::qCompare<int, unsigned int>(int const&, unsigned int const&, char const*, char const*, char const*, int) »
collect2: error: ld returned 1 exit status
tests/CMakeFiles/fontdatabasemodeltest.dir/build.make:176 : la recette pour la cible « bin/fontdatabasemodeltest » a échouée
make[2]: *** [bin/fontdatabasemodeltest] Erreur 1
make[2] : on quitte le répertoire « /src/KDAB/GammaRay/build »
CMakeFiles/Makefile2:1549 : la recette pour la cible « tests/CMakeFiles/fontdatabasemodeltest.dir/all » a échouée
make[1]: *** [tests/CMakeFiles/fontdatabasemodeltest.dir/all] Erreur 2
make[1] : on quitte le répertoire « /src/KDAB/GammaRay/build »
Makefile:160 : la recette pour la cible « all » a échouée
make: *** [all] Erreur 2
```

**=>** undefined reference to « bool QTest::qCompare<int, unsigned int>(int const&, unsigned int const&, char const*, char const*, char const*, int) »

**Comment by jgueytat** [comment]
It seems that modeltest.* files need the recent update done there:
http://code.qt.io/cgit/qt/qtbase.git/commit/?h=5.9&id=f99589a515cbeace0041a11abbcdf7f81a83d255

Compilation is getting further then.

**Comment by jgueytat** [comment]
Double checking.

**Comment by pasnox** [comment]
This issue is rather clear, 2 different types are used when calling QCOMPARE(), this need an explicit cast (or use the correct type at first).

**Comment by jgueytat** [comment]
Yes. That's exactly what is done over there:
http://code.qt.io/cgit/qt/qtbase.git/commit/tests/auto/other/modeltest/modeltest.cpp?h=5.9&id=6255cb893d411b055758f2e64e94fde0bce91ea8

**Comment by jgueytat** [comment]
https://github.com/KDAB/GammaRay/pull/317

**Comment by jgueytat** [comment]
Commit message fixed

**Comment by krf** [comment]
#317 got merged into 2.7 branch. Thanks!
